packages/query/src/index.ts

```diff
@@ -153,29 +154,23 @@ export class LiveQuery extends TxProcessor implements Storage, Queriable {
       const updatedDoc = q.result.find((p) => p._id === tx.objectId)
       if (updatedDoc !== undefined) {
         const pos = q.result.indexOf(updatedDoc)
-        await this.doUpdateDoc(updatedDoc, tx)
+        this.doUpdateDoc(updatedDoc, tx)
         if (!this.match(q, updatedDoc)) {
           q.result.splice(pos, 1)
           q.total--
         }
         this.sort(q, tx)
+        q.result[q.result.findIndex(p => p._id === updatedDoc._id)] = updatedDoc
```

Comment:
I think we need to sort after document is updated and you could use pod already defined